src/main/java/org/springframework/data/redis/connection/RedisClusterConfiguration.java

```diff
@@ -51,6 +52,8 @@ public class RedisClusterConfiguration implements RedisConfiguration, ClusterCon
 
 	private @Nullable Integer maxRedirects;
 
+	private @Nullable AsyncTaskExecutor executor;
```

Comment:
`RedisClusterConfiguration` doesn't seem ideal to hold the executor. Our `Redis…Configuration` classes are intended to describe the endpoint(s) to connect to and provide specific connectivity options such as the password or redirection limit that are used by the clients directly. 

`ClusterCommandExecutor` is a Spring Data concept. The alternative of moving the executor into `…ClientConfiguration` seems also not ideal because the executor isn't required by the client.

How about adding `executor` as property to `[Jedis|Lettuce]ConnectionFactory` directly? We have a few properties already, that are factory-specific (`convertPipelineAndTxResults`, `shareNativeConnection`, `pipeliningFlushPolicy`)? While the executor is only required for clustering operations, this change would give us the opportunity to revise its purpose along the lines of a "general purpose executor for asynchronous activity such as running multiple commands concurrently".

What do you think?

Reply:
+1 for moving the executor to the connection factory.